_test/test_sqw_class/test_smooth.m

```diff
@@ -0,0 +1,233 @@
+
+        function test_smooth_units_array_width_arg(obj)
+            s = sqw(obj.test_sqw_2d_fullpath);
+            s.data.pix = PixelData();
+
+            d = s.smooth_units([100, 25]);
+
+            % TODO: assert data matches expected...
+            %assertEqualToTol(d.data.s, expected.data.s, 1e-8);
+            %assertEqualToTol(d.data.e, expected.data.e, 1e-8);
+        end
+
+
+        function test_smooth_units_resolution_shape_arg(obj)
+            skipTest('No valid agruments possible for resolution call #628');
+            s = sqw(obj.test_sqw_2d_fullpath);
+            s.data.pix = PixelData();
+
+            d = s.smooth_units([1, 1.1, 1.2], 'resolution');
+
+            % TODO: assert data matches expected...
+            %assertEqualToTol(d.data.s, expected.data.s, 1e-8);
+            %assertEqualToTol(d.data.e, expected.data.e, 1e-8);
+        end
+
+        function test_smooth_units_hat_shape_arg(obj)
+            s = sqw(obj.test_sqw_2d_fullpath);
+            s.data.pix = PixelData();
+
+            d = s.smooth_units(100, 'hat');
+
+            % TODO: assert data matches expected...
+            %assertEqualToTol(d.data.s, expected.data.s, 1e-8);
+            %assertEqualToTol(d.data.e, expected.data.e, 1e-8);
+        end
+
+        function test_smooth_units_gaussian_shape_arg(obj)
+            s = sqw(obj.test_sqw_2d_fullpath);
+            s.data.pix = PixelData();
+
+            d = s.smooth_units(100, 'gaussian');
+
+            % TODO: assert data matches expected...
+            %assertEqualToTol(d.data.s, expected.data.s, 1e-8);
+            %assertEqualToTol(d.data.e, expected.data.e, 1e-8);
+        end
+
+        function test_smooth_units_raises_error_with_no_width_arg(obj)
+            s = sqw(obj.test_sqw_2d_fullpath);
+            s_nopix = copy(s);
+            s_nopix.data.pix = PixelData();
+
+            assertExceptionThrown(@() s_nopix.smooth_units(), 'HORACE:smooth:invalid_arguments');
+        end
+
+        function test_smooth_units_raises_error_with_invalid_shape_arg(obj)
+            s = sqw(obj.test_sqw_2d_fullpath);
+            s_nopix = copy(s);
+            s_nopix.data.pix = PixelData();
+
+            assertExceptionThrown(@() s_nopix.smooth_units(100, 'invalid_shape'), 'HORACE:smooth:invalid_arguments');
+        end
+
+        function test_smooth_units_raises_error_with_wrong_dimension_width_arg(obj)
+            s = sqw(obj.test_sqw_2d_fullpath);
+            s_nopix = copy(s);
+            s_nopix.data.pix = PixelData();
+
+            assertExceptionThrown(@() s_nopix.smooth_units([100, 123], 'invalid_shape'), 'HORACE:smooth:invalid_arguments');
+        end
+
+        function test_smooth_units_raises_error_if_any_sqw_has_pixel_data(obj)
+            s = sqw(obj.test_sqw_2d_fullpath);
+            s_nopix = copy(s);
+            s_nopix.data.pix = PixelData();
+
+            assertExceptionThrown(@() smooth_units([s_nopix, s, s_nopix], 10), 'HORACE:smooth:invalid_arguments');
```

Comment:
failure is not due to invalid arguments, it's due to second element of vector having pix data, please fix the error message